### DISCUSSION TITLE
Fastest way to get size of many images

### DISCUSSION THREAD

**Discussion by nik684** [ImageMagick/ImageMagick#7266] [Help] [answered]
### ImageMagick version

7.1.1-30

### Operating system, version and so on

Windows 10

### Description

Hello. I didn't find the answer here, but maybe I searched not well enough.

My task is to get resolution (size, i.e. width and height in px) of many images located in different folders. I need this data for my scripted tool. By simply using scripting language I can run magick command line and also run .bat file, but I'm facing performance issues. To get image data I use "magick identify -ping".

Files could be pretty large, could be located in different folders, those folders can contain other files that I don't need. Because of these reasons I need to specify a full file path (which is pretty long) for each file. I can't just use a long list of file paths in one command run because of chars limitation. This ends up with using loops, either in script itself or in bat file. This loop takes a lot more time to process in comparison to simple "magick identify -ping folder\*.* >> output.log" command.

The question is what other options I left? Maybe API could do the job by using some Python code or it won't help? Maybe I'm already missing something?

Thanks for your time.

**Comment by snibgo** [comment, marked as answer]
>  I can't just use a long list of file paths in one command run because of chars limitation.

What limitation? I assume a Windows limitation on command length.

I suggest using a list of the files in a text file, and referring to that file with the `@` notation. For example, using Windows, we make a file named `xlist.txt` containing one filename per line:
```
dir *.png /b >xlist.txt
```
Then:
```
magick identify -ping -format "%f %wx%h\n" @xlist.txt
```
This shows results like:
```
x000009.png 200x200
x000010.png 200x200
x1.png 1040x976
x2.png 1040x976
```
For escapes %f etc, see https://imagemagick.org/script/escape.php. If doing this in a BAT script file, double each percent `%%`.

If we want the filename, width and height in environment variables, we can do this:
```
setlocal enabledelayedexpansion

for /F "usebackq" %%L in (`magick identify -ping -format "Name=%%f\nWidth=%%w\nHeight=%%h\n" @xlist.txt`) do (
  set %%L
  {Do other processing using !Name!, !Width! and !Height!}
)
```
On my laptop, this takes about 0.4 seconds for 400 files.

**Reply by nik684** [reply]
Nice. I didn't know about @file.txt. Thanks a lot. It is much faster now.
